### PR TITLE
Add logs cmd loki-env

### LINKED ISSUE #11: logs command
add `logs` command to fetch logs for a given time window and component from loki

### PR DESCRIPTION
**Pull request by jfortin-sap-zz**
**What this PR does / why we need it**:
add logs command to fetch logs for a given time window and component from loki
**Which issue(s) this PR fixes**:
Fixes #11 

**Special notes for your reviewer**:

**Release note**:

```feature user
add logs command to fetch logs for a given time window and component from loki
```

### CODE REVIEW

File: pkg/cmd/env/options.go [hunk 1]
```diff
@@ -123,6 +128,24 @@ func (o *options) Run(f util.Factory) error {
...
+		if !t.ControlPlane() {
+			return target.ErrNoControlPlaneTargeted
+		}
+
+		client, err := manager.SeedClient(ctx, t)
+		if err != nil {
+			return fmt.Errorf("failed to create seed cluster client: %w", err)
+		}
+
+		shootns := "shoot--" + o.CurrentTarget.ProjectName() + "--" + o.CurrentTarget.ShootName()
```
Comment (grolu) [inline pkg/cmd/env/options.go:145]: use TechnicalID of shoot

File: pkg/cmd/env/options.go [hunk 2]
```diff
@@ -123,6 +128,24 @@ func (o *options) Run(f util.Factory) error {
...
+
+	case "loki":
+		t := o.CurrentTarget.WithControlPlane(true)
+		ctx := f.Context()
+
+		if !t.ControlPlane() {
+			return target.ErrNoControlPlaneTargeted
+		}
+
+		client, err := manager.SeedClient(ctx, t)
```
Comment (petersutter) [inline pkg/cmd/env/options.go:140]:
```
(⎈ |garden--gcp-external:default)❯ eval $(gardenctl loki-env zsh)
Error: failed to create seed cluster client: no Auth Provider found for name "gcp"
```
this happened when targeting a managed seed and it's seed was a soil cluster that was running on gke

**Review by petersutter — COMMENTED** [review]
please also add an alias in the rc command, similar to `gk`, maybe `gl`?

**Comment by vlerenc** [comment]
What is this actually doing? There are different Lokis and they are not exposed with their query interface either. Can you share some details. Is that working? What is it doing? Building together some config?

**Comment by jfortin-sap** [comment]
@vlerenc It fetches/exposes required env. vars. to query cluster's loki instance from the cli using Grafana's loki cli (logcli)
```

% go run main.go loki-env zsh
export LOKI_ADDR='https://go-sre-poc--gctlv2-dev.ingress.gcp.seed.dev.k8s.ondemand.com/api/datasources/proxy/2/';
export LOKI_USERNAME=;
export LOKI_PASSWORD=;

# Run this command to configure loki for your shell:
# eval $(gardenctl loki-env zsh)
```

**Comment by vlerenc** [comment]
OK, thanks @jfortin-sap. Yes, I am also using the data sources and access Loki via Grafana. So it's setting the env vars for shoots then. Got it!

Just a few comments:
- There is also the seed observability stack and it contains interesting data as well
- I found `logcli` to have issues and abort retrievals in case of networking issues (no retry possibly to configure)
- The main difficulty is anyway/actually pulling the right logs/knowing the right labels to select the logs for the different components, but that's a difficult subject matter

**Pull request closed without merging**